### PR TITLE
delete() calls to a report file will move it into sdcard storage

### PR DESCRIPTION
**Pull request by crankycoder**
This is a patch to get #1255 in so that delete() calls to a report file will actually move it into sdcard storage.

Most of the code here is actually robolectric related test code.  Important bits to note are:
- There is a test that exercises the checkbox in the DeveloperActivityFragment and setting the pref
- There is a test that makes sure that the SDCARD is mounted
  - the sdcard may be in an unmounted state, but robolectric will let you write to it anyway.  There doesn't seem to be a good way to test or mock this out.
- There is a test to verify the toast string
- There is a test to make sure files are moved from the report directory to the sdcard archive path.

### CODE REVIEW

File: android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import android.app.AlertDialog;
 import android.content.DialogInterface;
 import android.os.Bundle;
+import android.os.Environment;
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java:10]: not needed

File: android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java [hunk 2]
```diff
@@ -13,23 +14,24 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.AdapterView;
 import android.widget.CheckBox;
 import android.widget.CompoundButton;
-import android.widget.Spinner;
 import android.widget.TextView;
+import android.widget.Toast;
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java:25]: not needed

File: android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java [hunk 3]
```diff
@@ -13,23 +14,24 @@
...
 import org.mozilla.mozstumbler.client.serialize.KMLFragment;
 import org.mozilla.mozstumbler.service.AppGlobals;
-import org.mozilla.mozstumbler.service.core.logging.Log;
-import org.mozilla.mozstumbler.service.core.logging.MockAcraLog;
+import org.mozilla.mozstumbler.service.Prefs;
+import org.mozilla.mozstumbler.service.stumblerthread.datahandling.DataStorageManager;
+
+import java.io.File;
+
+import static org.mozilla.mozstumbler.R.string.create_log_archive_failure;
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/client/subactivities/DeveloperActivity.java]: 28-30 not needed

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java [hunk 4]
```diff
@@ -20,6 +25,8 @@
 import java.util.Timer;
 import java.util.TimerTask;
 
+import static org.mozilla.mozstumbler.R.string.create_log_archive_failure;
+
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java:22]: No resources in the service per favor

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java [hunk 5]
```diff
@@ -227,6 +239,34 @@ private String getStorageDir(Context c) {
         return dir.getPath();
     }
 
+
+    public static boolean archiveDirCreatedAndMounted(Context ctx) {
+        File saveDir = new File(sdcard_archive_path());
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java]: Can we pull this archiveDirCreatedAndMounted out to a client-side interface. maybe a subclass. Same with sdcard_archive_path, which should be sdcardArchivePath

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java [hunk 6]
```diff
@@ -513,14 +538,117 @@ public void writeSyncStats(long time, long bytesSent, long totalObs, long totalC
...
+                outStream.close();
+            } catch (IOException ioEx) {
+                Log.e(LOG_TAG, "error shutting down streams during a failed copy", ioEx);
+                ok = false;
+            }
+        }
+
+        // delete the original file
+        ok = ok & aFile.delete();
+        return ok;
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java]: return ok && aFile.delete()?

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java [hunk 7]
```diff
@@ -513,14 +538,117 @@ public void writeSyncStats(long time, long bytesSent, long totalObs, long totalC
         }
     }
 
+    /* Pass filename returned from dataToSend() */
+    public synchronized boolean delete(String filename) {
```
Comment (garvankeeley) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java:485]: Can we pull out this new logic to a client-side subclass?

**Comment by garvankeeley** [comment]
We should go over how are these files can become useful after not deleting them.

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java (hunk 5 above)
Comment (crankycoder) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java]: I've shuffled these bits into the DeveloperActivity now.

File: android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java (hunk 6 above)
Comment (crankycoder) [inline android/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/DataStorageManager.java]: doh

**Comment by crankycoder** [comment]
Hrm.  Subclassing DataStorageManager is a bit tricky, I'm going to need your help with this one because of the static createInstance calls.

**Comment by crankycoder** [comment]
This patch is ready, but it conflicts with dev - I'm going  to hold out on doing the 3-way merge until the low battery patches land.

**Pull request closed without merging**